### PR TITLE
[NavBar] Remove ENABLE_NEW_NAV_BAR branch

### PR DESCRIPTION
**Pull request by damassi**
Now that the new nav bar has been deployed to all users, this removes the `ENABLE_NEW_NAV_BAR` branch and old code. 

#### Todo
- [ ] Run `hokusai staging env unset ENABLE_NEW_NAVBAR`
- [ ] Run `hokusai production env unset ENABLE_NEW_NAVBAR`

### CODE REVIEW

**Comment by mzikherman** [comment]
Sweet! Some legit spec carnage, since the new nav shows up for realz need to stub the stitch component.

Amazing!!

File: src/desktop/components/main_layout/header/templates/index.jade [hunk 1]
```diff
@@ -1,12 +1,4 @@
 include ../../../util/activator
 
-if sd.ENABLE_NEW_NAVBAR
-  header
-    != stitch.components.NavBar({ notificationCount: sd.NOTIFICATION_COUNT, searchQuery: sd.searchQuery, user: sd.CURRENT_USER,  })
-else
-  header#main-layout-header( class= sd.HEADER_CLASS ? sd.HEADER_CLASS : '' )
-    if sd.APPLICATION_NAME === "force-staging"
```
Comment (mzikherman) [inline src/desktop/components/main_layout/header/templates/index.jade]: I'm ok with dropping this staging banner (I never really understood it?).

**Comment by joeyAghion** [comment]
Sort of surprised there isn't more styling to remove (all I see is a little CSS for the mobile drop-down). What am I missing?

File: src/desktop/components/main_layout/header/templates/index.jade (hunk 1 above)
Comment (damassi) [reply src/desktop/components/main_layout/header/templates/index.jade]: Staging banner is now in a  different component

**Comment by damassi** [comment]
@joeyAghion unless it's in a folder / file that can be clearly deleted it's not worth the effort / danger to hunt down global css properties for a few bytes.

**Comment by joeyAghion** [comment]
You're saying CSS is effectively... permanent? 😨 

I do not know the state of the art, but I'm pretty sure it will only get _harder_ than it is right now to clean old stylings up. Maybe there's some middle path (snapshot tests, unused CSS finder, ...)?

**Comment by damassi** [comment]
I understand the concern, but i assure you in this case it's ok 😄

**Comment by joeyAghion** [comment]
I'm breathing into a paper bag over here.

**Pull request merged**